django/middleware/csrf.py

```diff
@@ -136,6 +139,33 @@ class CsrfViewMiddleware(MiddlewareMixin):
     This middleware should be used in conjunction with the {% csrf_token %}
     template tag.
     """
+    @cached_property
+    def csrf_trusted_origins_hosts(self):
+        return [
+            urlparse(origin).netloc.lstrip('*')
+            for origin in settings.CSRF_TRUSTED_ORIGINS
+        ]
+
+    @cached_property
+    def allowed_origins_exact(self):
+        return {
+            origin for origin in settings.CSRF_TRUSTED_ORIGINS
+            if '*' not in origin
+        }
+
+    @cached_property
+    def allowed_origin_subdomains(self):
+        """
+        A mapping of allowed schemes to list of allowed netlocs, where all
+        subdomains of the netloc are allowed.
+        """
+        allowed_origin_subdomains = defaultdict(list)
+        for origin in settings.CSRF_TRUSTED_ORIGINS:
+            if '*' in origin:
+                parsed = urlparse(origin)
```

Comment:
Feel free to ignore this, but I noticed this can be tightened up as:

```python
for parsed in (urlparse(origin) for origin in settings.CSRF_TRUSTED_ORIGINS if '*' if origin):
```

Reply:
I'm okay with it. Hopefully @felixxm doesn't ding me for the longish line. I contemplated this version to shorten it but it doesn't seem as readable:
```python
parsed_origins = (
    urlparse(origin) for origin in settings.CSRF_TRUSTED_ORIGINS
    if '*' in origin
)
for parsed in parsed_origins:
```